File: AD-HOC/PYTHON/1026_1147/1031.py
def josephus(n, k):
    retorno = 0

    for i in range(1, n):
        retorno = (retorno + k) % i

    return retorno


def main():
    true_value = 1
    false_value = 0

    while true_value == 1:
        qts_regioes = int(input())

        if qts_regioes == 0:
            break

        pulo = 1
        while true_value == 1:
            if josephus(qts_regioes, pulo) != 11:
                pulo += 1
            else:
                break

        print(pulo)


if __name__ == "__main__":
    main()

File: AD-HOC/C#/TELA_1/1031.cs
using System;

class Program
{
    static int Josephus(int n, int k)
    {
        int retorno = 0;

        for (int i = 1; i < n; ++i)
            retorno = (retorno + k) % i;

        return retorno;
    }

    static void Main()
    {
        const int trueValue = 1;
        const int falseValue = 0;

        int qtsRegioes, pulo;

        while (trueValue == 1)
        {
            qtsRegioes = int.Parse(Console.ReadLine());

            if (qtsRegioes == 0)
                break;

            pulo = 1;
            while (trueValue == 1)
            {
                if (Josephus(qtsRegioes, pulo) != 11)
                    pulo++;
                else
                    break;
            }

            Console.WriteLine(pulo);
        }
    }
}
